scripts/aspects_migration.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+// utility for reading in the entire curriculum from a file path
+const fs = require('fs')
+const Insight = require('../lib/insight')
+const childProcess = require('child_process').execSync
+
+let curriculumPath
+let topic
+let curriculum
+
+const aspects = ['new', 'workout', 'deep', 'introduction', 'obscura']
+
+function loadCurriculumFromLocalPath () {
+  process.argv.forEach((val, index) => {
```

Comment:
instead of iterating over this, you can directly destructure it with something like this:
```
const [, , curriculumPath, topicName] = process.argv;
// ignores what's at index 0 and 1 and gets what's at index 2 => assigns to curriculumPath, gets what's at index 3 => assigns to topicName
```
then you can check if the `curriculumPath` is valid and throw an error otherwise:
```
if (!fs.existsSync(curriculumPath)) {
 throw new Error('Invalid curriculum Path')
}
```

Moreover, you can check if the `topicName` is valid as well:
```
if (!fs.existsSync(path.join(curriculumPath, topicName)) {
   throw new Error('Invalid topic name')
}
```
Subsequently, you can also enforce this behavior: if no `topicName` is provided, just do the migration for all the topics. However, given we'll use this just for one topic at the time, this is not that important to implement.

You can do all of these before calling `migrateAspectsFromTags()`, in the global scope.